### ISSUE TITLE
Tacho: Tacho doesn't declare dependency on rocblas properly

### ISSUE THREAD

**Issue by masterleinad** [trilinos/Trilinos#13636]
## Bug Report
Configuring `Trilinos` with `HIP` support but without `TPL_ROCBLAS` compiles but at runtime, I'm seeing
```
symbol lookup error: /tmp/trilinos-install/lib64/libtacho.so.16: undefined symbol: rocblas_create_handle
```
which indicates that `Tacho` is using `rocblas` without declaring a dependency on it. E.g., https://github.com/trilinos/Trilinos/blob/8bded3a266b3aa18b5f9fd4f380b5136982b3c23/packages/shylu/shylu_node/tacho/src/impl/Tacho_NumericTools_LevelSet.hpp#L558-L565 looks suspicious.

**Comment by iyamazaki** [comment]
Thank you, @masterleinad. Just to make sure I understand, when you enable ROCBLAS, ROCSPARSE, and ROCSOLVER, the issue with running Tacho goes away? And,it sounds like these are available for including and linking for compiling and building, but the symbols cannot be resolved at runtime with the dynamically-linked libraries?

**Comment by masterleinad** [comment]
> Just to make sure I understand, when you enable ROCBLAS, ROCSPARSE, and ROCSOLVER, the issue with running Tacho goes away? 

Yes, the issue goes away when those TPLs are enabled (since that implicitly also enables `ShyLU_NodeTacho_ENABLE_ROC*` and the respective preprocessor variables get defined).

> And,it sounds like these are available for including and linking for compiling and building, but the symbols cannot be resolved at runtime with the dynamically-linked libraries?

Yes, the API interface for `rocblas` seems to be available without enabling the corresponding TPL. It seems the include directories coming with the compiler are sufficient. The symbols are missing when not explicitly linking against the `rocblas` library.
Again, I tried just linking against it in the downstream application but ran into memory access issues on the GPU that only got resolved after also enabling the TPLs in `Trilinos` (which makes manually linking against those TPLs downstream obsolete).
